cellprofiler_core/readers/gcs_reader.py

```diff
@@ -0,0 +1,81 @@
+import logging
+import os
+from urllib.parse import urlparse
+
+import imageio
+from cellprofiler_core.readers.imageio_reader import ImageIOReader
+from google.cloud import storage
+
+from ..reader import Reader
+
+SUPPORTED_EXTENSIONS = {'.tiff'}
+
+
+class GcsReader(ImageIOReader):
+  """ Reads images from bucket(s) in Google Cloud Storage via user's Application Default Credential (ADC).
+  Leverages ImageIOReader image processing methods.
+
+  Prerequisites:
+  1. User has authenticated with Google Cloud Storage Application Default Credential (ADC) by running command, `gcloud auth application-default login`.
```

Comment:
Consider adding just a bit more to this prerequisite since there are several environments where the necessary credential is already configured. 
> 1. User has authenticated with Google Cloud Storage Application Default Credential (ADC) by running command, `gcloud auth application-default login`, or is running CellProfiler in an environment where this credential has already been configured for them such as app.terra.bio or CloudShell.

Reply:
I have updated the comment accordingly.